Libraries/CustomComponents/ListView/ListView.js

```diff
@@ -342,7 +348,16 @@ var ListView = React.createClass({
       var sectionID = dataSource.sectionIdentities[sectionIdx];
       var rowIDs = allRowIDs[sectionIdx];
       if (rowIDs.length === 0) {
-        continue;
+        var warning = require('fbjs/lib/warning');
+        if (this.props.enableEmptySections === undefined) {
+          warning(false, 'In next release empty section headers will be rendered. In this release you can use \'enableEmptySections\' flag to render empty section headers.');
+          continue;
+        } else {
+          warning(false, 'In next release \'enableEmptySections\' flag will be deprecated, empty section headers will always be rendered. If those headers are not desirable their indices should be excluded from sectionIDs object.');
+          if (!this.props.enableEmptySections) {
```

Comment:
We shouldn't support false since we want to remove that behavior. You can use an invariant here to make sure, and include a helpful error message explaining that the prop will be deprecated.